src/main/cpp/RLBotInterface/RLBotInterface/GameFunctions/PlayerInfo.cpp

```diff
@@ -92,21 +92,34 @@ namespace GameFunctions
 	// Ctypes
 	extern "C" RLBotCoreStatus RLBOT_CORE_API UpdatePlayerInput(PlayerInput playerInput, int playerIndex)
 	{
-		RLBotCoreStatus status = checkInputConfiguration(playerInput);
+		flatbuffers::FlatBufferBuilder builder;
+		FlatbufferTranslator::inputStructToFlatbuffer(&builder, playerInput, playerIndex);
+		RLBotCoreStatus status = UpdatePlayerInputFlatbuffer(builder.GetBufferPointer(), builder.GetSize());
 
 		if (status != RLBotCoreStatus::Success)
 			return status;
 
-		flatbuffers::FlatBufferBuilder builder;
-		FlatbufferTranslator::inputStructToFlatbuffer(&builder, playerInput, playerIndex);
-		return UpdatePlayerInputFlatbuffer(builder.GetBufferPointer(), builder.GetSize());
+		// We're validating the input *after* sending it to the core dll because the core dll
+		// is going to clamp it for us with no issues. We're reporting validation errors to
+		// users just for their information because it might help them discover faulty logic.
+		return checkInputConfiguration(playerInput);
```

Comment:
Seems a little weird to me to validate it afterwards but I guess it makes sense for reporting errors to the user